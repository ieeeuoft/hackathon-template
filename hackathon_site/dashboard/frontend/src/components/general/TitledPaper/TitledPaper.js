import React from "react";
import styles from "./TitledPaper.module.scss";
import Typography from "@material-ui/core/Typography";
import Paper from "@material-ui/core/Paper";

const TitledPaper = ({ title, children }) => (
    <>
        {title && (
            <Typography variant="h2" noWrap className={styles.title}>
                {title}
            </Typography>
        )}
        <Paper elevation={3} square={true} className={styles.paper}>
            {children}
        </Paper>
    </>
);

export default TitledPaper;
