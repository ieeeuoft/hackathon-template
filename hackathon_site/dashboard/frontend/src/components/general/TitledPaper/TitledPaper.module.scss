@import "./../../../assets/abstracts/variables";

.title {
    margin-bottom: 10px;
}

.paper {
    flex-grow: 1;
    padding: 8px 0px;
    svg {
        color: color(grey);
    }
}
