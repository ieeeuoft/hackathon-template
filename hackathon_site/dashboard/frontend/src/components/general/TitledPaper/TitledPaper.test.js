import React from "react";
import { render } from "@testing-library/react";
import TitledPaper from "./TitledPaper";

describe("<TitledPaper />", () => {
    const titleTest = "Test title";
    const childTest = "Test child";

    it("renders text given to it and title appears", () => {
        const { getByText, getByTestId } = render(
            <TitledPaper title={titleTest}>{childTest}</TitledPaper>
        );
        expect(getByTestId("titledPaperTitle")).toBeInTheDocument();
        expect(getByText(titleTest)).toBeInTheDocument();
        expect(getByText(childTest)).toBeInTheDocument();
    });
    it("does not render title if not given anything", () => {
        const { queryByTestId } = render(<TitledPaper />);
        expect(queryByTestId("titledPaperTitle")).not.toBeInTheDocument();
    });
});
