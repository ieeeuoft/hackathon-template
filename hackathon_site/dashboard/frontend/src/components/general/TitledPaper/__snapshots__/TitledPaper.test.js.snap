// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamCard matches snapshot 1`] = `
<DocumentFragment>
  <h2
    class="MuiTypography-root title MuiTypography-h2 MuiTypography-noWrap"
  >
    Test title
  </h2>
  <div
    class="MuiPaper-root paper MuiPaper-elevation3"
  >
    Test child
  </div>
</DocumentFragment>
`;
